### PR TITLE
Simulator event worker fix

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/cache/gossip.js [hunk 1]
```diff
@@ -85,12 +86,18 @@ exports.notifyFlag = async ({ledgerNodeId, remove = false, add = false}) => {
...
   const key = _cacheKey.gossipPeerStatus({creatorId, ledgerNodeId});
   return cache.client.multi().hmset(
-    key, 'backoff', backoff, 'lastContactDate', lastContactDate,
-    'lastContactResult', lastContactResult)
+    key,
+    'backoff', backoff,
+    'detectedBlockHeight', detectedBlockHeight,
+    'lastContactDate', lastContactDate,
+    'lastContactResult', lastContactResult
+  )
```
Comment (dlongley) [inline lib/cache/gossip.js:100]:
```suggestion
    'lastContactResult', lastContactResult);
```

File: lib/events.js [hunk 2]
```diff
@@ -1413,3 +1485,161 @@ async function _getStartHash(
   const eventHash = await getStartHash({creatorId, targetGeneration});
   return {eventHash, truncated: true};
 }
+
+/*
+  LOOKING FOR A HOME
```
Comment (dlongley) [inline lib/events.js:1490]: Needs some attention (either a new home, or a decision that it doesn't need one with whatever reworking is necessary).

File: lib/events.js [hunk 3]
```diff
@@ -1413,3 +1485,161 @@ async function _getStartHash(
...
+  blockHeight, events, ledgerNode, needed
+}) {
+  const ledgerNodeId = ledgerNode.id;
+
+  // FIXME: temporary fix for clearing completed jobs
+  // clears jobs that were *completed* more than 5 seconds ago
+  // no need to await this
+  jobQueue.clean(5000);
+
+  // FIXME: use removeOnComplete when fixed
```
Comment (dlongley) [inline lib/events.js:1525]: FIXMEs need attention

**Review by dlongley — COMMENTED** [review]
I was asked to review this but it looks like it's actually not quite ready. Left some comments.

File: lib/events.js (hunk 2 above)
Comment (mattcollier) [reply lib/events.js:1490]: Leaving this for now by consensus.

**Review by dlongley — APPROVED** [review]

**Pull request merged**